nitro-protocol/contracts/CurrencySwapApp.sol

```diff
@@ -0,0 +1,156 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+pragma experimental ABIEncoderV2;
+
+import './interfaces/IForceMoveApp.sol';
+import './libraries/NitroUtils.sol';
+import './interfaces/INitroTypes.sol';
+import {ExitFormat as Outcome} from '@statechannels/exit-format/contracts/ExitFormat.sol';
+
+/**
+ * @dev The CurrencySwapApp contract complies with the ForceMoveApp interface and allows payments to be made virtually from Alice to Bob (participants[0] to participants[n+1], where n is the number of intermediaries).
+ */
+contract CurrencySwapApp is IForceMoveApp {
+
+    struct VoucherSwap {
+        address takerAsset; // TODO: support multi-chain with struct {chainId, Address}
+        uint256 takerAmount;
+        address makerAsset;
+        uint256 makerAmount;
+        INitroTypes.Signature signature; // signature on abi.encode(channelId, amounts)
+        // TODO: I assume both side need to confirm the amount
+        // INitroTypes.Signature takerSignature;
+        // INitroTypes.Signature makerSignature;
+    }
+
+    //TODO: Remove once logic is tested
+    struct VoucherAmountAndSignature {
+        uint256 amount;
+        INitroTypes.Signature signature; // signature on abi.encode(channelId,amount)
+    }
+
+    // TODO: To confirm indices are always 0 (Taker) and 1 (Maker)
+    enum AllocationIndices {
+        Taker,  // Aggressor order
+        Maker   // Passive order was found in broker orderbook
```

Comment:
You are the expert on this, but I would have thought the Maker proposes a trade, and the Taker fulfils it? What you have seems the other way around. 

In any case I think it is fine, we just have to be a) consistent and b) track down any assumptions of unidirectionality and remove them (see above comment about replacing vouchers).

Reply:
To be exact, Maker has an offer online, ready to be taken, but the initiator is what we call the Aggressing order, which is the one that will trigger the trade (the price is matching)